src/Todo/Model.hs

```diff
@@ -32,3 +53,33 @@ data Todo = Todo
     , todoDeadline    :: Maybe Deadline
     , todoDescription :: Maybe String
     } deriving (Eq, Show)
+
+instance Hashable Todo where
+    hashWithSalt :: Int -> Todo -> Int 
+    hashWithSalt s Todo {..} = 
+        s `hashWithSalt` 
+        todoState `hashWithSalt` 
+        todoTitle `hashWithSalt` todoPriority
+
+defTodo :: Title -> Priority -> Tags -> Todo
+defTodo todoTitle todoPriority todoTags = Todo { todoState = TODO
+                                               , todoSubtasks = HSet.empty
+                                               , todoDeadline = Nothing
+                                               , todoDescription = Nothing
+                                               , .. }
+
+getByState :: State -> Todos -> Todos
+getByState state = HSet.filter $ (state ==) . todoState
+
+getByTag :: Tag -> Todos -> Todos
+getByTag tag = HSet.filter $ NESet.member tag . todoTags
+
+getTopPrio :: Int -> Todos -> Todos
+getTopPrio n = HSet.fromList . take n . sortBy (flip $ comparing todoPriority) . HSet.toList
+
+tagStatistics :: Todos -> [(Tag, Int)]
+tagStatistics = count . concatMap (NESet.toList . todoTags)
+  where
+    -- from Data.List.Unique
+    count :: Ord a => [a] -> [(a, Int)]
+    count = map (\x -> (head x, length x)) . group . sort
```

Comment:
Okay, finally it's time to implement `count` function efficiently using `HashMap` and in more type-safe way. `length` for list is a very slow function. It's better to have mapping from key to counter. Also, `group` function actually has type `[a] -> [NonEmpty a]` so you should use this function from `Data.List.NonEmpty` module:

* https://hackage.haskell.org/package/base-4.11.0.0/docs/Data-List-NonEmpty.html#v:group